plugin.video.catchuptvandmore/resources/lib/resolver_proxy.py

```diff
@@ -200,28 +318,36 @@ def get_brightcove_video_json(plugin,
                               data_account,
                               data_player,
                               data_video_id,
-                              download_mode=False):
+                              policy_key=None,
+                              download_mode=False,
+                              subtitles=None):
+    if policy_key is None:
+        # Method to get JSON from 'edge.api.brightcove.com'
+        key = get_brightcove_policy_key(data_account, data_player)
+    else:
+        key = policy_key
 
-    # Method to get JSON from 'edge.api.brightcove.com'
-    resp = urlquick.get(
-        URL_BRIGHTCOVE_VIDEO_JSON % (data_account, data_video_id),
-        headers={
-            'User-Agent':
-            web_utils.get_random_ua(),
-            'Accept':
-            'application/json;pk=%s' %
-            (get_brightcove_policy_key(data_account, data_player)),
-            'X-Forwarded-For':
-            plugin.setting.get_string('header_x-forwarded-for')
-        })
-    json_parser = json.loads(resp.text)
+    headers = {
+        'User-Agent': web_utils.get_random_ua(),
+        'Accept': 'application/json;pk=%s' % key,
+        'X-Forwarded-For': plugin.setting.get_string('header_x-forwarded-for')
```

Comment:
We don't want add-ons to circumvent GEO blocks, both in streaming and downloading. That means that using `x-forwarded-for` is not allowed.